src/librustc_resolve/lib.rs

```diff
@@ -1441,13 +1441,13 @@ impl<'a> Resolver<'a> {
     }
 
     /// Entry point to crate resolution.
-    pub fn resolve_crate(&mut self, krate: &Crate) {
+    pub fn resolve_crate(&mut self, krate: &Crate, ignore_bodies: bool) {
         let _prof_timer = self.session.prof.generic_activity("resolve_crate");
 
         ImportResolver { r: self }.finalize_imports();
         self.finalize_macro_resolutions();
 
-        self.late_resolve_crate(krate);
+        self.late_resolve_crate(krate, ignore_bodies);
```

Comment:
This could be an `enum`, I was recently bitten by a ~~radioactive~~ `bool` function input that I thought was the opposite of what it was.

Also, this could be done in a couple other ways:
* detect `rustdoc` from the `Session` (probably not great?)
* set it when creating the `Resolver`, as opposed to customizing it here

Although, if `rustdoc` isn't in control of the creation of the `Resolver` then only the former is a viable alternative.

Reply:
> detect rustdoc from the Session (probably not great?)

This is how it gets set in the first place: https://github.com/rust-lang/rust/pull/73566/files#diff-24c5c945888bb0d041e769bfb852de6cR413. Not sure it makes sense to move that further into `rustc_resolve`, but I can if you think it limits the scope of the hack.

> set it when creating the Resolver, as opposed to customizing it here

I thought resolvers were used for both early and late resolution? I want to only ignore late errors, macros need to resolve properly no matter what.

Reply:
I changed the boolean to an enum. It reuses the existing enum so the docs are a little iffy but the variant names make sense.